alicloud/table_alicloud_slb_load_balancer.go

```diff
@@ -0,0 +1,336 @@
+				Name:        "account_id",
+				Description: ColumnDescriptionAccount,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getCommonColumns,
+				Transform:   transform.FromField("AccountID"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listSlbLoadBalancers(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create service connection
+	client, err := SLBService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("alicloud_slb_load_balancer.listSlbLoadBalancers", "connection_error", err)
+		return nil, err
+	}
+
+	request := slb.CreateDescribeLoadBalancersRequest()
+	request.Scheme = "https"
+	request.PageSize = requests.NewInteger(50)
+	request.PageNumber = requests.NewInteger(1)
+
+	if d.KeyColumnQualString("load_balancer_name") != "" {
+		request.LoadBalancerName = d.KeyColumnQualString("load_balancer_name")
+	}
+	if d.KeyColumnQualString("network_type") != "" {
+		request.NetworkType = d.KeyColumnQualString("network_type")
+	}
+	if d.KeyColumnQualString("resource_group_id") != "" {
+		request.ResourceGroupId = d.KeyColumnQualString("resource_group_id")
+	}
+	if d.KeyColumnQualString("master_zone_id") != "" {
+		request.MasterZoneId = d.KeyColumnQualString("master_zone_id")
+	}
+	if d.KeyColumnQualString("address_ip_version") != "" {
+		request.AddressIPVersion = d.KeyColumnQualString("address_ip_version")
+	}
+	if d.KeyColumnQualString("v_switch_id") != "" {
+		request.VSwitchId = d.KeyColumnQualString("v_switch_id")
+	}
+	if d.KeyColumnQualString("vpc_id") != "" {
+		request.VpcId = d.KeyColumnQualString("vpc_id")
+	}
+	if d.KeyColumnQualString("load_balancer_status") != "" {
+		request.LoadBalancerStatus = d.KeyColumnQualString("load_balancer_status")
+	}
+	if d.KeyColumnQualString("address_type") != "" {
+		request.AddressType = d.KeyColumnQualString("address_type")
+	}
+	if d.KeyColumnQualString("internet_charge_type") != "" {
+		request.InternetChargeType = d.KeyColumnQualString("internet_charge_type")
+	}
+
+	count := 0
+	for {
+		response, err := client.DescribeLoadBalancers(request)
+		if err != nil {
+			plugin.Logger(ctx).Error("alicloud_slb_load_balancer.listSlbLoadBalancers", "api_error", err, "request", request)
+			return nil, err
+		}
+		for _, loadBalancer := range response.LoadBalancers.LoadBalancer {
+			d.StreamListItem(ctx, loadBalancer)
+			count++
+		}
+		if count >= response.TotalCount {
+			break
+		}
+		request.PageNumber = requests.NewInteger(response.PageNumber + 1)
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getSlbLoadBalancer(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := d.KeyColumnQualString(matrixKeyRegion)
+
+	// Create service connection
+	client, err := SLBService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("alicloud_slb_load_balancer.getSlbLoadBalancer", "connection_error", err)
+		return nil, err
+	}
+
+	var id string
+	if h.Item != nil {
+		id = databaseID(h.Item)
```

Comment:
@ParthaI why is this check needed? I don't see any column needing this hydrate call explicitly.

I think we should also add an empty value check for the `load_balancer_id` column.

Reply:
Updated it